src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -95,6 +95,9 @@ export class QuickKillTerminalAction extends Action {
 		if (terminal) {
 			terminal.dispose();
 		}
+		if (this.terminalService.activeTerminalInstanceIndex !== terminalIndex) {
+			this.terminalService.setActiveInstanceByIndex(Math.min(terminalIndex, this.terminalService.terminalInstances.length - 1));
```

Comment:
Is it a problem if there's only one terminal, and you delete it? Does it call this with `-1`?